mesa/experimental/altair_grid.py

```diff
@@ -0,0 +1,67 @@
+import json
+from typing import Callable, Optional
+
+import altair as alt
+import solara
+
+import mesa
+
+
+def get_agent_data_from_coord_iter(data):
+    for agent, (x, y) in data:
+        if agent:
+            agent_data = json.loads(
+                json.dumps(agent[0].__dict__, skipkeys=True, default=str)
+            )
+            agent_data["x"] = x
+            agent_data["y"] = y
+            agent_data.pop("model", None)
+            agent_data.pop("pos", None)
+            yield agent_data
+
+
+def create_grid(
+    color: Optional[str] = None,
+    on_click: Optional[Callable[[mesa.Model, mesa.space.Coordinate], None]] = None,
+) -> Callable[[mesa.Model], solara.component]:
+    return lambda model: Grid(model, color, on_click)
```

Comment:
Please rewrite this to remove the lambda expression; in general we want to try and keep lambda functions out of the mesa base code to make it more readable